weed/s3api/s3api_circuit_breaker.go

```diff
@@ -103,14 +115,17 @@ func (cb *CircuitBreaker) Limit(f func(w http.ResponseWriter, r *http.Request),
 			f(w, r)
 			return
 		}
+
 		s3err.WriteErrorResponse(w, r, errCode)
+		stats.S3RequestLimitCounter.WithLabelValues(action).Inc()
 	}, Action(action)
 }
 
-func (cb *CircuitBreaker) limit(r *http.Request, bucket string, action string) (rollback []func(), errCode s3err.ErrorCode) {
+func (cb *CircuitBreaker) limit(r *http.Request, bucket string, action string) (limitAction string, rollback []func(), errCode s3err.ErrorCode) {
```

Comment:
how about create a struct as
```
type TrafficLimit struct{
  count int64
  size int64
}
```

This way can avoid creating the concatenated the keys for each request, and are more type-safe.

Reply:
looks good, i will try it

Reply:
> how about create a struct as
> 
> ```
> type TrafficLimit struct{
>   count int64
>   size int64
> }
> ```
> 
> This way can avoid creating the concatenated the keys for each request, and are more type-safe.



> how about create a struct as
> 
> ```
> type TrafficLimit struct{
>   count int64
>   size int64
> }
> ```
> 
> This way can avoid creating the concatenated the keys for each request, and are more type-safe.



> how about create a struct as
> 
> ```
> type TrafficLimit struct{
>   count int64
>   size int64
> }
> ```
> 
> This way can avoid creating the concatenated the keys for each request, and are more type-safe.

..., I didn't understand your idea, I'm curious how to find the corresponding limit configuration based on bucket, action, limitType without concatenating keys.

Reply:
..., I didn't understand your idea, I'm curious how to find the corresponding limit configuration based on bucket, action, limitType without concatenating keys @chrislusf

Reply:
What I'm thinking of is to put `limitation` and `count` in a struct `TrafficLimit`, so that finding the struct by key directly completes the operation (increasing or decrementing the counter) without having to look up the counter again